src/sysinfo.cc

```diff
@@ -46,6 +46,12 @@
 #if defined(BENCHMARK_OS_QURT)
 #include <qurt.h>
 #endif
+#if defined(BENCHMARK_HAS_PTHREAD_AFFINITY)
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
```

Comment:
is this necessary?  if it's just for pthread can we revert the definition afterwards please?  i'd like to avoid people accidentally depending on non-standard extensions or non-portable functions.